core/src/com/unciv/logic/automation/unit/WorkerAutomation.kt

```diff
@@ -155,6 +162,13 @@ class WorkerAutomation(
             return
         }
 
+        //If we have reached a fort tile that is in progress and shouldn't be there, cancel it.
+        if ((currentTile.improvementInProgress == Constants.fort
```

Comment:
Double parentheses again

Reply:
Why do we need this twice in the same function? Looks the same as line 134

Reply:
The lines 134-154 don't make sense now that I look at them more closely. Here is my interpretation of what it was trying to do, but I don't agree with what I perceive to be the goal of it.
If a worker moves from a tile that does not have any improvements scheduled (mostly from running away) and ends up moving to it's destination tile without improving anything on the tile. Then the worker will improve the tile that it came from.
This might not be desired behavior because manually controlled workers cannot repeat this functionality.

Line 134 has reachedTile == tileToWork which doesn't make sense to me and the inconsistent use of currentTile and reachedTile in 135 and 136 are confusing me.

If we want to keep this functionality we will need to remove the changes I made replacing currentTile with reachedTile. However I recommend reverting the replacements and fixing this in another commit because it is not relevant to the purpose of this commit.

Reply:
Between the first and second batches there is no worker movement going on.
I think we can keep the first (line 135) so that other improvements can be scheduled, and delete the second (line 166) because it will always be false

Reply:
Agreed, it seems to be working without line 164-170, logically it makes sense as well that it is unnecessary.